libcontainer/integration/init_test.go

```diff
@@ -19,12 +23,40 @@ func init() {
 	}
 	runtime.GOMAXPROCS(1)
 	runtime.LockOSThread()
+
+	logLevel := logrus.InfoLevel
+	if level := os.Getenv("_LIBCONTAINER_LOGLEVEL"); level != "" {
+		var err error
+		logLevel, err = logrus.ParseLevel(level)
+		if err != nil {
+			panic(fmt.Sprintf("failed to parse _LIBCONTAINER_LOGLEVEL=%s: %s", level, err))
+		}
+	}
+
+	logPipeFdStr := os.Getenv("_LIBCONTAINER_LOGPIPE")
+	logPipeFd, err := strconv.Atoi(logPipeFdStr)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse _LIBCONTAINER_LOGPIPE=%s: %s", logPipeFdStr, err))
+	}
+	err = logs.ConfigureLogging(logs.Config{
+		LogPipeFd: logPipeFd,
+		LogFormat: "json",
+		LogLevel:  logLevel,
+	})
+	if err != nil {
+		panic(fmt.Sprintf("failed to configure logging: %v", err))
+	}
+	logrus.Debug("child process in init()")
+
+	// let ForwardLogs exit
+	unix.Close(logPipeFd)
+
 	factory, err := libcontainer.New("")
 	if err != nil {
-		logrus.Fatalf("unable to initialize for container: %s", err)
```

Comment:
`logrus` is replaced by panic, because we just closed the logrus file descriptor. This is a test case so doesn't matter much.